src/process_utility.c

```diff
@@ -309,6 +311,137 @@ process_reindex(Node *parsetree)
 }
 
 
+static void
+process_altertable_change_owner(Cache *hcache, AlterTableCmd *cmd, Oid relid)
+{
+	Hypertable *ht;
+	RoleSpec   *role;
+
+	ht = hypertable_cache_get_entry(hcache, relid);
+	if (NULL == ht)
+		return;
+
+	Assert(IsA(cmd->newowner, RoleSpec));
+	role = (RoleSpec *) cmd->newowner;
+	OidFunctionCall2(catalog_get_internal_function_id(catalog_get(), DDL_CHANGE_OWNER),
```

Comment:
All the constraint.c stuff in your PR is way less readable than simple sql functions. In PLPGSQL it's way easier to see directly what's going on. Rather than putting together a bunch of nodes to execute functions. My way is also orders of magnitude more compact.